src/Talis/Persona/Client/Base.php

```diff
@@ -119,21 +112,26 @@ public function __construct(array $config)
      */
     public function getStatsD()
     {
-        if ($this->statsD == null) {
+        if (is_null($this->statsD)) {
             $connStr = getenv(self::STATSD_CONN);
-            if (!empty($connStr)) {
-                list($host, $port) = explode(":", $connStr);
-                $conn = new \Domnikl\Statsd\Connection\Socket($host, $port);
+
+            if (!empty($connStr) && !empty(strpos($connStr, ':'))) {
+                list($host, $port) = explode(':', $connStr);
+                $conn = new UdpSocket($host, $port);
```

Comment:
TODO: UdpSocket is a new class that belongs to the StatsD library. Which version does RL use? Is this going to cause any issues? @kiyanwang would you know?

Reply:
AFAIK RL doesn't directly use the StatsD Library

There are places in the code where RL sends stats through Echo i.e. The TalisEcho wrapper class in Tarl is used for this.

Tripod-php sends stats directly to statsd https://github.com/talis/tripod-php/blob/master/src/classes/StatsD.class.php#L89-L129  but as you can see this is handrolled and does not make use of that library